openedx_tagging/core/tagging/models.py

```diff
@@ -456,3 +490,95 @@ def resync(self) -> bool:
             changed = True
 
         return changed
+
+
+class OpenObjectTag(ObjectTag):
+    """
+    Free-text object tag.
+
+    Only needs a free-text taxonomy and a value to be valid.
+    """
+
+    class Meta:
+        proxy = True
+
+    def _check_taxonomy(self):
+        """
+        Returns True if this ObjectTag has a valid taxonomy.
+
+        Subclasses should override this method to perform any additional validation for the particular type of object tag.
+        """
+        # Must be linked to a free-text taxonomy
+        return self.taxonomy_id and self.taxonomy.allow_free_text
+
+    def _check_tag(self):
+        """
+        Returns True if this ObjectTag has a valid tag value.
+
+        Subclasses should override this method to perform any additional validation for the particular type of object tag.
+        """
+        # Open taxonomies don't need an associated tag, but we need a value.
+        return bool(self._value)
+
+    def _check_object(self):
+        """
+        Returns True if this ObjectTag has a valid object.
+
+        Subclasses should override this method to perform any additional validation for the particular type of object tag.
+        """
+        # Must have a valid object id/type:
+        return self.object_id and self.object_type
```

Comment:
@ChrisChV So, I haven't yet found a use for this `ObjectTag.object_type` field..

* The ObjectTag subclasses that require UsageKey/CourseKey formatting for `course_id` can get the object type from the object key.
* We'll be creating a sytem-defined Taxonomy which automatically creates ObjectTags to mark "Content Type".

However, we might need a way to associate Taxonomy instances with the various object types they support. E.g, the [system-defined Language Taxonomy may be used on Course, Block, Library, and Library-type content items](https://github.com/open-craft/openedx-learning/pull/3#discussion_r1258942677). So when we're in the Course editor and we ask the API "What taxonomies are available here?", we'll need to know which ones are enabled for use with Courses.

I don't know if this is required for MVP though, since I think all the taxonomies discussed will be used by all the different content types. But Taxonomy Admins might want to create some that are only used for Blocks, for example.

So if we do need to support that, I'd propose using a many-to-many mapping like we did with [TaxonomyOrg](https://github.com/openedx/edx-platform/pull/32661/files#diff-8a644422ddd022128d13b6ca7c8b672a33d048d1816a80a80491246bdd76f607R13), and extending the [`get_taxonomies` API](https://github.com/openedx/openedx-learning/pull/62/files#diff-951a5c4d32e1301ac1c30419ebce4ada8ae17db6c93f4c2a5a34caac9eb30e0aR55) to support querying by object type.

Reply:
@pomegranited It's a great idea!. On the [MVP requeriments ](https://openedx.atlassian.net/wiki/spaces/OEPM/pages/3695247491/Content+Tagging+Approach+MVP#Features-%26-Requirements) there is:

> Course Settings: Content Authors can turn taxonomies (instance and org-levels) on/off at the course level.

I feel that your proposal can also be adapted to that requirement. But I feel like it would have to be a many-to-many relationship with [TaxonomyOrg](https://github.com/openedx/edx-platform/pull/32661/files#diff-8a644422ddd022128d13b6ca7c8b672a33d048d1816a80a80491246bdd76f607R13) and extend the [get_taxonomies_for_org](https://github.com/openedx/edx-platform/pull/32661/files#diff-8a644422ddd022128d13b6ca7c8b672a33d048d1816a80a80491246bdd76f607R13).